File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

long long solution(long long n) {
    long long answer = -1;
    double d = sqrt(n);
    long long l = (long long)d;
    if (d == (double)l) {
        answer = (d+1)*(d+1);
    }
    return answer;
} 

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution {
    public long solution(long n) {
        long answer = 0;
        double a = Math.sqrt((double)n);
        long b = Math.round(a);
        if(a != (double)b) {
            answer = -1;
        } else {
            answer = (b+1)*(b+1);
        }
        return answer;
    }
}